### PR TITLE
[KotlinCleanup] AsyncDialogFragment.kt

### PR DESCRIPTION
**Pull request by Jatin887**
## Purpose / Description
Kotlin CleanUp
## Fixes
Part of #10489
## Approach
Using lint and some code changes which were mentioned in the comments and required to make the code more reliable

## How Has This Been Tested?
Physical Device One Plus Nord2

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Jatin887** [comment]
@mikehardy please review

**Comment by Jatin887** [comment]
> Seems like exactly the prescribed cleanup to me thanks!
> 
> Two things about workflow -
> - you don't need to tag me, or anyone for review, there are a few of us that scan the queue any time we have time available - just set the labels correctly if possible, or be patient. Sometimes I just don't have time at which point a notification from a tag is actually a negative for me
> - you need to familiarize yourself with the github markdown issue linking with regards to keyword "fixes", "related" etc and how to correctly cross-link an issue, I think I've pointed it out on your PRs before and fixed them but you have generated a few that are not really cross-linked correctly. I'm not fixing this one for you, I want to see you edit the description and fix it (it's easy I promise)

I'll keep these points in mind for my future pull requests.
Thank you :)

**Comment by mikehardy** [comment]
The description edit looks perfect :-), thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt [hunk 1]
```diff
@@ -19,20 +19,19 @@ import android.content.res.Resources
...
-    @KotlinCleanup("convert these back to properties")
-    abstract fun getNotificationMessage(): String?
-    abstract fun getNotificationTitle(): String?
-    open fun getDialogHandlerMessage(): Message? {
-        return null
-    }
+    abstract val notificationMessage: String?
+    abstract val notificationTitle: String?
+    open val dialogHandlerMessage: Message?
+        get() {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt:32]: style: I'd prefer a single line for this: 

```
open val dialogHandlerMessage: Message? get() = null
```

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt [hunk 2]
```diff
@@ -394,22 +394,25 @@ class DatabaseErrorDialog : AsyncDialogFragment() {
             else -> res().getString(R.string.answering_error_title)
         }
 
-    override fun getNotificationMessage(): String? {
-        return message
-    }
+    override val notificationMessage: String?
+        get() {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:398]: ditto - single line for brevity.

**Review by david-allison — COMMENTED** [review]
seems good, one request around making the functions less verbose

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt (hunk 1 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt]: adeed changes

**Review by david-allison — APPROVED** [review]

**Pull request merged**